spec/controllers/api/encryptables_transfer_controller_spec.rb

```diff
@@ -75,4 +76,64 @@
     end
   end
 
+  context 'Encryptable Credentials Transfer' do
+
+    it 'Does not send encryptable credentials to non existing user' do
+      login_as(:bob)
+
+      request_params = {
+        receiver_id: 139082421,
+        encryptable_id: credentials2.id
+      }
+
+      expect do
+        post :create, params: request_params, xhr: true
+      end.to raise_error(ActiveRecord::RecordNotFound, "User with id 139082421 does not exist")
+
+    end
+
+    it 'Bob sends credentials to Alice' do
+      login_as(:bob)
+
+      request_params = {
+        receiver_id: alice.id,
+        encryptable_id: credentials2.id
+      }
+
+      post :create, params: request_params, xhr: true
+
+      expect(response).to have_http_status(200)
+
+      transferred_encryptable = alice.personal_team.folders.last.encryptables.last
+      plaintext_team_password =
+        alice.personal_team.decrypt_team_password(alice, alice.decrypt_private_key('password'))
+
+      EncryptableTransfer.new.receive(transferred_encryptable,
+                                      alice.decrypt_private_key('password'),
+                                      plaintext_team_password)
+
+      expect(transferred_encryptable.name).to eq('Twitter Account')
+      expect(transferred_encryptable.id).not_to eq(credentials2.id)
+      expect(transferred_encryptable.description).to eq('My personal twitter account')
+      expect(transferred_encryptable.sender_id).to eq(bob.id)
+      expect(transferred_encryptable.encrypted_transfer_password).to eq(nil)
+    end
+  end
+
+  it 'Does not send encryptable credentials which the sender does not have access to' do
+    login_as(:alice)
+    api_controller = ApiController.new
+
+    request_params = {
+      receiver_id: bob.id,
+      encryptable_id: credentials2.id
+    }
+
+    # There is no error raised when user don't have access
```

Comment:
well, then raise an error in that case ! 
you probably need a special pundit policy for this case.
so `authorize entry` should fail in that case